src/coreclr/tools/superpmi/superpmi-shim-collector/icorjitinfo.cpp

```diff
@@ -968,6 +968,19 @@ bool interceptor_ICJI::isMoreSpecificType(CORINFO_CLASS_HANDLE cls1, CORINFO_CLA
     return temp;
 }
 
+// Returns TypeCompareState::Must if cls is known to be an enum.
+// For enums with known exact type returns the underlying
+// type in underlyingType when the provided pointer is
+// non-NULL.
+// Returns TypeCompareState::May when a runtime check is required.
+TypeCompareState interceptor_ICJI::isEnum(CORINFO_CLASS_HANDLE cls, CORINFO_CLASS_HANDLE* underlyingType)
+{
+    mc->cr->AddCall("isEnum");
+    TypeCompareState temp = original_ICorJitInfo->isEnum(cls, underlyingType);
+    mc->recIsEnum(cls, underlyingType, temp);
```

Comment:
Optional out args are a bit tricky in SPMI.

Since the key here is just the class handle, if a the jit invokes this method multiple times with a mixture of null and non-null `underlyingType` args, we run the risk of not recording the underlying type, ad each successive call updates the map with its results.

Can you modify this so that it always asks for and stores the underlying type, then only set the `underlyingType` out arg if non-null?

Something like the following:

```suggestion
    CORINFO_CLASS_HANDLE tempUnderlyingType;
    TypeCompareState temp = original_ICorJitInfo->isEnum(cls, &tempUnderlyingType);  
    mc->recIsEnum(cls, tempUnderlyingType, temp);
    if (underlyingType != nullptr)
    {
         *underlyingType = tempUnderlyingType;
    }
```